spec/acceptance/class_spec.rb

```diff
@@ -16,13 +16,16 @@
     service_name = 'rabbitmq'
   end
 
+  repos_ensure = case fact('os.family')
+                 when 'RedHat'
+                   true
+                 end
```

Comment:
So the current behavior of the module IIRC is that `repos_ensure` defaults to `false`. Changing it for the acceptance test here means (I think) that this will now not test the module’s default behavior.

there were some discussions of changing that default again in the module itself, but I think that would be more involved, and that’s where previous attempts at this have kind of stalled in the absence of an active maintainer.

Reply:
#845 has some good discussion and I think is a good starting point.

Reply:
I think the module should provide sane and working defaults for the different operating systems and the tests should at least cover the defaults. If we test non-default values as well that's a nice addition.

Reply:
I understand what you mentioned but even for CentOS7 currently this module can't provide complete configuration. This still relies on puppet-eralang to enable epel, now.

Reply:
Also, AFAIK Ubuntu doesn't require the additional repo and can install rabbitmq from usual OS repos. If we enable repos then default repos will be replaced, which I don't think would be ideal setup.

Reply:
Why not enable `repos_ensure` only for RedHat? Since as it seems only RedHat needs extra repos to work out of the box.
That's a common pattern IMO

Reply:
This module has never set up the required repositories by default. The current implementation relies on epel repository enabled externally. This change still keeps the same policy and let the users to enable the repository setup by their own(any external modules or setting this parameter.)
I initially hesitated to enable this because it can break existing usage(for example we use this module to deploy OpenStack but in that case we use rabbitmq packages provided by RDO, not packagecloud). However if all agrees we should enable this and make the all users to handle this change properly then I can enable it.

Reply:
@kajinamit this module did, at one point, when Puppet still maintained it, setup the RabbitMQ repo by default. It also had an option to include the simple distribution of erlang that rabbitmq uses.

For better or worse, I am responsible for making that change back then (#493). At the time, it seemed like a sane deafult. I think with hindsight, going back to the old behavior (with `repos_ensure` being default again, as a breaking change) is probably where things should go, it's just a matter of getting there when this project doesn't have an active maintainer.